src/drtsans/transmission.py

```diff
@@ -124,10 +150,26 @@ def calculate_transmission(input_sample, input_reference, radius, radius_unit="m
     # Notify of incorrect calculation of zero angle transmission
     # Will happen if the beam centers have been totally masked
     if bool(np.all(np.isnan(zero_angle_transmission_workspace.readY(0)))) is True:
-        raise RuntimeError("Transmission at zero-angle is NaN")
+        raise TransmissionNanError("Transmission at zero-angle is NaN")
 
-    # Notify of average transmission value
     non_gap_indexes = np.isfinite(zero_angle_transmission_workspace.readY(0))
+
+    if transmission_error_tolerance:
+        # Verify that errors are below the transmission error tolerance
+        transmission_relative_error = (
+            zero_angle_transmission_workspace.readE(0)[non_gap_indexes]
+            / zero_angle_transmission_workspace.readY(0)[non_gap_indexes]
+        )
+        if not np.all(transmission_relative_error < transmission_error_tolerance):
+            i_max = np.argmax(transmission_relative_error)
+            error_max = transmission_relative_error[i_max]
+            error_max_transmission = zero_angle_transmission_workspace.readY(0)[non_gap_indexes][i_max]
+            raise TransmissionErrorToleranceError(
+                f"Transmission error {error_max} > transmission error tolerance "
```

Comment:
I don't understand this error message, which is why I requested to include a check for the expected error message in `test_transmission.py::test_transmission_error_tolerance`